pkgs/applications/networking/cluster/argocd-cli/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib, buildGoPackage, fetchFromGitHub, packr }:
+
+buildGoPackage rec {
+  pname = "argocd-cli";
+  version = "1.3.0";
+
+  goPackagePath = "github.com/argoproj/argo-cd";
+
+  goDeps = ./deps.nix;
+
+  src = fetchFromGitHub {
+    owner = "argoproj";
+    repo = "argo-cd";
+    rev = "v${version}";
+    sha256 = "0g3nfrdls61wqa2p2xp9vlbqlm7kk4llbzqqydk3vkr99i2whvml";
+  };
+ 
+  nativeBuildInputs = [ packr ];
+ 
+  buildPhase = ''
+    # remove nested dependencies
+    rm -rf go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery
+    rm -rf go/src/k8s.io/kubernetes/vendor/k8s.io/api
+
+    cd go/src/${goPackagePath}
+    packr
+    go build cmd/argocd/main.go
+  '';
+
+  installPhase = ''
+    install -D main $bin/bin/argocd
+  '';
+ 
+  meta = with lib; {
+    description = "Argo CD is a declarative, GitOps continuous delivery tool for Kubernetes";
+    homepage = https://github.com/argoproj/argo;
+    license = licenses.asl20;
+    maintainers = with maintainers; [ shahrukh330 ];
+    platforms = platforms.unix;
```

Comment:
```suggestion
```

Already set to [`platforms.all`](https://github.com/NixOS/nixpkgs/blob/d217d4c63c4acecceb12153c285014e65bd8409d/pkgs/development/go-packages/generic/default.nix#L242) which is a fit for most Go binaries.

Reply:
Updated